plugin-build/src/main/kotlin/io/sentry/android/gradle/util/tasks.kt

```diff
@@ -0,0 +1,89 @@
+package io.sentry.android.gradle.util
+
+import io.sentry.android.gradle.SentryTasksProvider.capitalized
+import io.sentry.android.gradle.SentryTasksProvider.getAssembleTaskProvider
+import io.sentry.android.gradle.SentryTasksProvider.getBundleTask
+import io.sentry.android.gradle.SentryTasksProvider.getPackageBundleTask
+import io.sentry.android.gradle.SentryTasksProvider.getPackageProvider
+import io.sentry.android.gradle.SentryTasksProvider.getPreBundleTask
+import io.sentry.android.gradle.SentryTasksProvider.getTransformerTask
+import io.sentry.android.gradle.util.SentryPluginUtils.withLogging
+import io.sentry.gradle.common.AndroidVariant
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.TaskProvider
+
+fun TaskProvider<out Task>.hookWithMinifyTasks(
+    project: Project,
+    variantName: String,
+    experimentalGuardsquareSupport: Boolean
+) {
+    // we need to wait for project evaluation to have all tasks available, otherwise the new
+    // AndroidComponentsExtension is configured too early to look up for the tasks
+    project.afterEvaluate {
+        val transformerTaskProvider = withLogging(project.logger, "transformerTask") {
+            getTransformerTask(
+                project,
+                variantName,
+                experimentalGuardsquareSupport
+            )
+        }
+
+        if (experimentalGuardsquareSupport &&
+            GroovyCompat.isDexguardEnabledForVariant(project, variantName)
```

Comment:
shouldn't this be an or?
```suggestion
        if (experimentalGuardsquareSupport ||
            GroovyCompat.isDexguardEnabledForVariant(project, variantName)
```